final-project/scripts/main.js

```diff
@@ -1,10 +1,110 @@
-import "../styles/main.scss";
-
-// testing that `const` and arrow functions transpile correctly
-const before = "Hello -world!";
-const after = before
-  .split("")
-  .filter(char => char !== "-")
-  .join("");
-console.log({ before });
-console.log({ after });
+import "../styles/main.scss";
+import Axios from "axios";
+
+function showPage() {
+  document.getElementById("body-layout").classList.toggle("layout-wrapper--loading");
+  document.getElementById("page-loader").style.display = "none";
+}
+
+window.onload = function () {
+  //Wait just to simulate a low connection and a delay
+  setTimeout(showPage, 3000);
+
+  //Retrieve blog posts
+  const url = 'https://wt-4662f45b9eefda7172b747b28d23efdb-0.sandbox.auth0-extend.com/blog';
+  Axios
+    .get(url)
+    .then(data => {
+      const articles = data.data.articles;
+      for (let i = 0; i < articles.length; i++) {
+        const article = articles[i];
+
+        // Add articles
+        $('#main-blog-articles')
+          .append(`
+            <li id="blog-article-${i+2}" class="main-blog-carousel-slides__slide">
+              <img src="${article.images.mobile}" />
+              <article>
+                <h1>${article.title}</h1>
+                <p>${article.description}</p>
+                <div class="main-blog-controls">
+                  <button><a href="${article.url}" target="_blank">Read now</a></button>
+                  <span><a>Add to your bookmarks</a></span>
+                </div>
+              </article>
+            </li>
+          `);
+        
+        // Add navigation circles
+        $('#main-blog-carousel')
+          .append(`
+          <li class="carousel__li"><i id="carousel-blog-${i+2}" class="fas fa-circle"></i></li>
+          `);
+      }
+
+      // Click on a carousel circle, show blog's article index
+      $('#main-blog-carousel>li>i').click(event => {

```

Comment:
Depending on DOM structure for selectors is not a good practice, elements can be changed and break the functionality